frontend/src/components/SimpleSelect.tsx

```diff
@@ -33,13 +33,18 @@ export type SimpleSelectOption = {
 export type SimpleGroupSelectOption = {
   key: string;
   label: string;
-  options: SimpleSelectOption[];
+  options: SimpleSelectOptionStrict[];
 };
 
+// Prevents addition of anything outside what's defined in SimpleSelectOption
+type Strict<T> = T & { [K in Exclude<keyof T, keyof SimpleSelectOption>]: never };
+
+export type SimpleSelectOptionStrict = Strict<SimpleSelectOption>;
+
 type SimpleSelectProps = {
-  options?: SimpleSelectOption[];
+  options?: SimpleSelectOptionStrict[];
   groupedOptions?: SimpleGroupSelectOption[];
-  value?: string;
+  value: string | undefined;
```

Comment:
very minor nitpick, but all the other properties for this type are using `...?: ` and not `...: .. | undefined`. So i'd personally prefer this to not be changed. (The 2 different styles are quite subjective though, i just match what exists usually)
```suggestion
  value?: string;
```